regex.c

```diff
@@ -248,58 +227,55 @@ void read_regex_from_file(void)
 	if(config.debug & DEBUG_REGEX)
 		regexbuffer = calloc(num_regex, sizeof(char*));
 
-	// Search through file
-	// getline reads a string from the specified file up to either a
-	// newline character or EOF
-	for(int i=0; getline(&buffer, &size, fp) != -1; i++)
+	// Connect to whitelist table
+	if(!gravityDB_getTable(REGEX_LIST))
+	{
+		logg("read_regex_from_database(): Error getting table from database");
+		return;
+	}
+
+	// Walk database table
+	const char *domain = NULL;
+	int i = 0;
+	while((domain = gravityDB_getDomain()) != NULL)
 	{
-		// Test if file has changed since we counted the lines therein (unlikely
-		// but not impossible). If so, read only as far as we have reserved memory
+		// Avoid buffer overflow if database table changed
+		// since we counted its entries
 		if(i >= num_regex)
 			break;
 
-		// Strip potential newline character at the end of line we just read
-		if(buffer[strlen(buffer)-1] == '\n')
-			buffer[strlen(buffer)-1] = '\0';
-
 		// Skip this entry if empty: an empty regex filter would match
 		// anything anywhere and hence match (and block) all incoming domains.
 		// A user can still achieve this with a filter such as ".*", however
 		// empty lines in regex.list are probably not expected to have such an
```

Comment:
Comment should be changed as regexes are loaded from the database instead of a file now.